helix-term/src/ui/popup.rs

```diff
@@ -116,7 +116,15 @@ impl<T: Component> Component for Popup<T> {
                 self.scroll(self.size.1 as usize / 2, false);
                 EventResult::Consumed(None)
             }
-            _ => self.contents.handle_event(event, cx),
+            _ => {
+                let contents_event_result = self.contents.handle_event(event, cx);
+
+                if let EventResult::Ignored(None) = contents_event_result {
```

Comment:
only applies if `EventResult::Ignored(None)`, to prevent cases where inner contents event result could have a callback that executes something like `compositor.pop()`